src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -325,7 +325,7 @@ public override object Clone()
                     continue;
                 }
 
-                send.Inventory.Add(inv);
+                send.Inventory.Add(new InventoryVector(peer.AddSupportedOptions(InventoryType.MSG_TX), inv.Hash));
```

Comment:
Why not 

```
inv.Type = peer.AddSupportedOptions(InventoryType.MSG_TX);
send.Inventory.Add(inv);
```

to avoid creating a new instance of `InventoryVector`

Reply:
I'm not sure I like that. I prefer to keep collections immutable especially with async stuff going on, but maybe I'm just paranoid. Should be noted InventoryVector is only a hash and flag.

Reply:
This collection is not used after that method so it's ok to modify items inside of it or even add/remove items from the collection. 

But ok, I don't mind.